webapp/app.py

```diff
@@ -101,3 +103,41 @@ def get_users(username: str):
         return jsonify(list(users))
     else:
         return jsonify({"error": "Failed to fetch users"}), 500
+
+
+@app.route('/set-owners', methods=['POST'])
+# @login_required
+def set_owners():
+    data = request.get_json()
+
+    user = data.get("user_struct")
+    webpage_id = data.get("webpage_id")
+    user_hrc_id = user.get("id")
+
+    # If user does not exist, create a new user in the "users" table
+    user_exists = User.query.filter_by(hrc_id=user_hrc_id).first()
+    if not user_exists:
+        user_exists, _ = get_or_create(db.session,
```

Comment:
is this checking the "users" table by all the fields? we should check only by "hrc_id", as other data can change over time, for example user's job title or department or team.

Reply:
In line 118 it is doing the filtering by checking `hrc_id` only